### ISSUE TITLE
Error sending email - sendEmailTemplate.groovy.setTextMsg

### ISSUE THREAD

**Issue by mrovnanik** [moqui/moqui-framework#128]
I customized my reset password screen (it uses ec.l10n.localize). But the problem seems to occur even with the original password screen.
![image](https://cloud.githubusercontent.com/assets/14863763/19066553/186c28e4-8a1a-11e6-9d67-9c737b71c3f2.png)

This is what I get when user resets his password:
![image](https://cloud.githubusercontent.com/assets/14863763/19066199/5a7e70ea-8a18-11e6-8a63-ab9f9116da4c.png)

![image](https://cloud.githubusercontent.com/assets/14863763/19066242/935c36ae-8a18-11e6-96f9-349eacff3cd0.png)

![image](https://cloud.githubusercontent.com/assets/14863763/19066697/d2ce5b8a-8a1a-11e6-93c5-92a8137a7d41.png)

**Comment by jonesde** [comment]
This may be due to recent changes in the text output, which is now designed for fixed width output for line printing and such. It may be that you have another error in your log related to this, most likely before this error.

**Comment by mrovnanik** [comment]
I printed out the whole log console output into file. Here it is:
[ErrorSendingMail.txt](https://github.com/moqui/moqui-framework/files/509504/ErrorSendingMail.txt)

**Comment by mrovnanik** [comment]
I implemented a "workaround" for now, is it risky?
![image](https://cloud.githubusercontent.com/assets/14863763/19142114/37e8af44-8b9b-11e6-916d-cf88c73ae5da.png)

**Comment by jonesde** [comment]
I don't know if 'risk' is the right word, it basically sends only the HTML part with no text part. This is generally okay as most email clients will handle the HTML message just fine and it may be fine for how you are using it, but not something that would make a good general default for the framework.

**Comment by jonesde** [comment]
There are a couple of fixes now in the moqui-framework and moqui-runtime repositories to resolve this issue:

- bodyHtml and bodyText are not added if empty (null or zero length)
- the DefaultScreenMacros.text.ftl macros have been fixed for cases like this, still fixed width but when outside a form use the full number of character columns; this isn't ideal for plain text output in emails like this, but works fine

To make testing easier, and as a nice general feature, the SubEtha SMTP server library is now included in moqui-framework (is nice and small on top of jars we are already using) and in the default configuration (see the EmailServer record for the MOQUI_LOCAL emailServerId) it listens on port 2525 for SMTP traffic and calls EMECA rules for emails received. There is authentication for it, an email.root user also configured on that EmailServer record and will authenticate any valid user in the system too for receiving email that gets processes with EMECA rules.

I tested it with the PasswordReset email and it is working fine.
